src/Input/Input.js

```diff
@@ -99,6 +99,11 @@ module.exports = kind(
 			var oThis = this;
 			util.asyncMethod(this, function () {oThis._bFocused = true;});
 		}
+		// Force cursor to end of text during a generic focus event. Creating the input by compiling
+		// a string of text with value="this.value" produces different initial caret position than
+		// using node.setAttribute('value', this.value), which is what would happen any time after
+		// the initial creation.
+		this.hasNode().selectionStart = this.value.length;
```

Comment:
I think maybe we should retain more of the original comment here, since the stated reason for the original change (to prevent the cursor from being in an untenable position in 5-way mode) seems more important than the consistent-caret-positioning reason stated in the new issue.